pkg/pxe/pxe.go

```diff
@@ -0,0 +1,258 @@
+			return err
+		}
+	}
+	return fmt.Errorf("no valid pxelinux config found")
+}
+
+// ParseConfigFile parses a PXE/Syslinux configuration as specified in
+// http://www.syslinux.org/wiki/index.php?title=Config
+//
+// Currently, only the APPEND, INCLUDE, KERNEL, LABEL, DEFAULT, and INITRD
+// directives are partially supported.
+//
+// `wd` is the default scheme, host, and path for any files named as a
+// relative path. The default path for config files is assumed to be
+// `wd.Path`/pxelinux.cfg/.
+func ParseConfigFile(uri string, wd *url.URL) (*Config, error) {
+	c := NewConfig(wd)
+	if err := c.AppendFile(uri); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// AppendFile parses the config file downloaded from `uri` and adds it to `c`.
+func (c *Config) AppendFile(uri string) error {
+	cfgWd := *c.wd
+	// The default location for looking for configuration is
+	// CWD/pxelinux.cfg/, so if `uri` is just a relative path, this will be
+	// the default directory.
+	cfgWd.Path = path.Join(cfgWd.Path, "pxelinux.cfg")
+
+	r, err := c.schemes.GetFile(uri, &cfgWd)
+	if err != nil {
+		return ErrConfigNotFound
+	}
+	config, err := ioutil.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	return c.Append(string(config))
+}
+
+// Append parses `config` and adds the respective configuration to `c`.
+func (c *Config) Append(config string) error {
+	// Here's a shitty parser.
+	for _, line := range strings.Split(config, "\n") {
+		// This is stupid. There should be a FieldsN(...).
+		kv := strings.Fields(line)
+		if len(kv) <= 1 {
+			continue
+		}
+		directive := strings.ToLower(kv[0])
+		var arg string
+		if len(kv) == 2 {
+			arg = kv[1]
+		} else {
+			arg = strings.Join(kv[1:], " ")
+		}
+
+		switch directive {
+		case "default":
+			c.DefaultEntry = arg
+
+		case "include":
+			if err := c.AppendFile(arg); err == ErrConfigNotFound {
+				// TODO: What to do in this case?
+				// Return for now. Test this with pxelinux.
+				return err
+			} else if err != nil {
+				return err
+			}
+
+		case "label":
+			// We forever enter label scope.
+			c.scope = scopeEntry
+			c.curEntry = arg
+			c.Entries[c.curEntry] = &Entry{}
+			c.Entries[c.curEntry].Cmdline = c.globalAppend
+
+		case "kernel":
+			k, err := c.schemes.LazyGetFile(arg, c.wd)
+			if err != nil {
+				return err
+			}
+			c.Entries[c.curEntry].Kernel = k
+
+		case "initrd":
+			i, err := c.schemes.LazyGetFile(arg, c.wd)
+			if err != nil {
+				return err
+			}
+			c.Entries[c.curEntry].Initrd = i
+
+		case "append":
+			switch c.scope {
+			case scopeGlobal:
+				c.globalAppend = arg
+
+			case scopeEntry:
+				if arg == "-" {
+					c.Entries[c.curEntry].Cmdline = ""
+				} else {
+					c.Entries[c.curEntry].Cmdline = arg
+				}
+			}
+		}
+	}
+
+	// Go through all labels and download the initrds.
+	for _, label := range c.Entries {
+		// If the initrd was set via the INITRD directive, don't
+		// overwrite that.
+		//
+		// TODO(hugelgupf): Is this really what syslinux does? Does
```

Comment:
This is why parsing these files is such a mess. The languages are highly improvised.

Reply:
Some day, when this doesn't work, we can go read ipxe source code or whatever. Until then, this is gonna be it :)